src/autotype/AutoType.cpp

```diff
@@ -374,6 +374,17 @@ void AutoType::performGlobalAutoType(const QList<QSharedPointer<Database>>& dbLi
         executeAutoTypeActions(matchList.first().entry, nullptr, matchList.first().sequence, m_windowForGlobal);
         m_inGlobalAutoTypeDialog.unlock();
     } else {
+        if (config()->get(Config::AutoTypeHideExpiredEntry).toBool()) {
+            auto it = matchList.begin();
+            while (it != matchList.end()) {
+                if (it->entry->isExpired()) {
+                    it = matchList.erase(it);
```

Comment:
You should be excluding the expired entries before they are even added to the list. This happens about 20 lines up.

Reply:
alright, yes this avoids unnecessary iterating over the list again. I added checks in the list build up loop.